manifests/register.pp

```diff
@@ -10,6 +10,7 @@
       effective_user  => $foreman_proxy::oauth_effective_user,
       ssl_ca          => pick($foreman_proxy::foreman_ssl_ca, $foreman_proxy::ssl_ca),
       url             => $foreman_proxy::real_registered_proxy_url,
+      instance_id     => $foreman_proxy::instance_id,
```

Comment:
It feels to me like there is some duplication. The instance ID is exposed in https://github.com/theforeman/smart-proxy/pull/783 but we also set it here.

Wouldn't it be better if Foreman automatically detected it rather than requiring it to be set beforehand?